src/backend/identities.rs

```diff
@@ -781,7 +933,34 @@ impl AppState {
         Ok(identity_result)
     }
 
-    pub(crate) async fn refresh_identity_balance(&mut self, sdk: &Sdk) -> Result<(), Error> {
+    pub(crate) async fn refresh_all_identities<'s>(&'s self, sdk: &Sdk) -> Result<(), Error> {
+        let mut known_identities = self.known_identities.lock().await;
+
+        let mut identities_to_refresh = Vec::new();
+
+        // Collect all known identity IDs to refresh
+        for identity in known_identities.values() {
+            identities_to_refresh.push(identity.id());
+        }
+
+        for identity_id in identities_to_refresh {
+            if let Some(refreshed_identity) = Identity::fetch(sdk, identity_id).await? {
+                known_identities
+                    .entry(identity_id)
+                    .and_modify(|id| *id = refreshed_identity.clone())
+                    .or_insert(refreshed_identity);
+            } else {
+                return Err(Error::IdentityRefreshError(format!(
```

Comment:
Maybe that's unnecessary, but "refresh all identities" sounds like it shall try to refresh all of them, but here it short-circuits on the first failure